src/rust/lib/graph-editor/src/component/visualization/class.rs

```diff
@@ -154,3 +176,126 @@ impl Visualization {
         self.state.renderer.set_size(size)
     }
 }
+
+
+
+// ===========================
+// === Visualization Class ===
+// ===========================
+
+/// Result of the attempt to instantiate a `Visualization` from a `Class`.
+pub type InstantiationResult = Result<Visualization,Box<dyn Error>>;
```

Comment:
If I'm correct, we are using different error handling method - please confirm it with @farmaazon and @mwu-tow before merging this PR. The way used here has 2 problems:

1. You cannot pattern match on the error after catching it.
2. The type `Box<dyn Error>` will appear in type errors during compilation because it is not a newtype, so users of this library can easily use it in their code etc - this makes the types in the app uglier.

Reply:
`Box<dyn Error>>` has been replaced with a proper error enum `InstantiationError`.